openff/interchange/interop/openmm/_topology.py

```diff
@@ -9,42 +9,50 @@
 
 
 def to_openmm_topology(
-    interchange: "Interchange", ensure_unique_atom_names: bool = True
+    interchange: "Interchange",
+    ensure_unique_atom_names: Union[str, bool] = "residues",
 ) -> "app.Topology":
     """Create an OpenMM Topology containing some virtual site information (if appropriate)."""
     # Heavily cribbed from the toolkit
     # https://github.com/openforcefield/openff-toolkit/blob/0.11.0rc2/openff/toolkit/topology/topology.py
 
     from collections import defaultdict
 
+    from openff.toolkit.topology import Topology
     from openff.toolkit.topology.molecule import Bond
     from openmm import app
 
     from openff.interchange.interop._virtual_sites import (
         _virtual_site_parent_molecule_mapping,
     )
 
-    topology = interchange.topology
+    # Copy topology to avoid modifying input (eg, when generating atom names)
+    topology = Topology(interchange.topology)
```

Comment:
In a perfect world I'd like to avoid copying large objects [^1] but I can see the value here. IIUC the only modifications happening here are tinkering with the atom names, but 
* I started thinking about what it would take to implement a mapping between the original and "modified as requested" atom names and could tell I was too lazy to see that idea to completion 
* It stands to reason there are more modifications that we'll be involved in in the future ...


[^1]: I have not done the profiling to understand what's really taking up space in memory - but I figure a `Topology` of a large protein in water will be large using our current object models